x-pack/metricbeat/module/prometheus/collector/histogram.go

```diff
@@ -23,35 +23,47 @@ import (
 //	}
 //
 // This code takes a Prometheus histogram and tries to accommodate it into an ES histogram by:
+//
 //   - calculating centroids for each bucket (values)
+//     - for +Inf "le" bucket, use the preceding bucket's value
+//     - for the first bucket only: if it has a negative "le", use the value as-is; otherwise use half its value
+//     - for all other buckets, use the midpoint from that bucket's value to the preceding bucket's
 //   - undoing counters accumulation for each bucket (counts)
+//     - `counts` is respresenting an array of rates, where rate of the first bucket is always 0, meaning that it
+// 		  was not increased as it is the first
 //
 // https://www.elastic.co/guide/en/elasticsearch/reference/master/histogram.html
```

Comment:
Maybe also link to the design/motivation issue here so as to have it as reference in the future.

Reply:
do you think the link to this comment https://github.com/elastic/apm-agent-python/pull/1165#discussion_r651397014 would be a good reference here?

Reply:
That would help yes, or at least give a pointer for anyone interested.

Reply:
done - https://github.com/elastic/beats/pull/36647/commits/4fd2e7fac748089afd49e8791ab7fa741ccf546b